src/oscar/apps/dashboard/offers/views.py

```diff
@@ -113,8 +113,7 @@ def _store_form_kwargs(self, form):
         form_data = form.cleaned_data.copy()
         range = form_data.get('range', None)
         if range is not None:
-            form_data['range_id'] = range.id
```

Comment:
Since the ID is saved rather than the object, the name might be confusing. I have a solution because I thought this might be down to personal taste.

If you wish, I can leave this part as it is but then on line 128-> I could do this

```
form_kwargs = json.loads(json_data)
if 'range_id' in form_kwargs['data']:
    form_kwargs['data']['range'] = form_kwargs['data']['range_id']
    del form_kwargs['data']['range_id']
    return form_kwargs
```